src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -162,20 +205,25 @@ public void close() throws IOException {
 
     @Override
     public long currentId() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return -1;
+        try (ReleasableLock _ = readLock.acquire()) {
+            return current.translogId();
         }
-        return current1.id();
     }
 
     @Override
     public int estimatedNumberOfOperations() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return 0;
+        int ops = 0;
+        try (ReleasableLock _ = readLock.acquire()) {
```

Comment:
see comment above. we have to protect the uncomittedTranslogs . An alternative is to cache these operations for all but current, but as I said, I think it's not worth it.